test/e2e/scheduler_priorities.go

```diff
@@ -0,0 +1,715 @@
+					},
+				},
+			},
+		})
+		By("Wait the pod becomes running")
+		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
+		labelPod, err := cs.Core().Pods(ns).Get(labelPodName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		By("Verify the pod was scheduled to the expected node.")
+		Expect(labelPod.Spec.NodeName).NotTo(Equal(nodeName))
+	})
+	It("Pod should avoid to schedule to node that have avoid annotation", func() {
+		nodeName := nodeList.Items[0].Name
+		// make the nodes have balanced cpu,mem usage
+		createBalancedPodForNodes(f, cs, ns, nodeList.Items, podRequestedResource, 0.5) //
+		By("Create a RC, with 0 replicas")
+		rc := createRC(ns, "scheduler-priority-avoid-pod", int32(0), map[string]string{"name": "scheduler-priority-avoid-pod"}, f, podRequestedResource)
+		// Cleanup the replication controller when we are done.
+		defer func() {
+			// Resize the replication controller to zero to get rid of pods.
+			if err := framework.DeleteRCAndPods(f.ClientSet, f.InternalClientset, f.Namespace.Name, rc.Name); err != nil {
+				framework.Logf("Failed to cleanup replication controller %v: %v.", rc.Name, err)
+			}
+		}()
+
+		By("Trying to apply avoid annotations on the first node.")
+		avoidPod := v1.AvoidPods{
+			PreferAvoidPods: []v1.PreferAvoidPodsEntry{
+				{
+					PodSignature: v1.PodSignature{
+						PodController: &metav1.OwnerReference{
+							APIVersion: "v1",
+							Kind:       "ReplicationController",
+							Name:       rc.Name,
+							UID:        rc.UID,
+							Controller: func() *bool { b := true; return &b }(),
+						},
+					},
+					Reason:  "some reson",
+					Message: "some message",
+				},
+			},
+		}
+
+		framework.AddOrUpdateAvoidPodOnNode(cs, nodeName, avoidPod)
+		framework.ExpectNodeHasAvoidPods(cs, nodeName, avoidPod)
+		defer framework.RemoveAvoidPodsOffNode(cs, nodeName)
+
+		By(fmt.Sprintf("Scale the RC: %s to len(nodeList.Iten)-1 : %v.", rc.Name, len(nodeList.Items)-1))
+
+		framework.ScaleRC(f.ClientSet, f.InternalClientset, ns, rc.Name, uint(len(nodeList.Items)-1), true)
+		testPods, err := cs.Core().Pods(ns).List(metav1.ListOptions{
+			LabelSelector: "name=scheduler-priority-avoid-pod",
+		})
+		framework.PrettyPrintJSON(testPods)
+		Expect(err).NotTo(HaveOccurred())
+		By(fmt.Sprintf("Verify the pods should not scheduled to the node: %s", nodeName))
+		for _, pod := range testPods.Items {
+			Expect(pod.Spec.NodeName).NotTo(Equal(nodeName))
+		}
+	})
+	It("Pod should perfer to scheduled to nodes pod can tolerate", func() {
+		// make the nodes have balanced cpu,mem usage
+		createBalancedPodForNodes(f, cs, ns, nodeList.Items, podRequestedResource, 0.5) //
+		//we need applay more taint on a node, because one match toleration only count 1
+		By("Trying to apply 10 taint on the nodes except first one.")
+		nodeName := nodeList.Items[0].Name
+
+		for index, node := range nodeList.Items {
+			if index == 0 {
+				continue
+			}
+			for i := 0; i < 10; i++ {
+				testTaint := v1.Taint{
```

Comment:
Optional: I would create a function to for this block, as it is repeated below and may be useful in the future, but up to you.